### DISCUSSION TITLE
Is it dangerous when I not use RLS when I am using supabase from serverside ?

### DISCUSSION THREAD

**Discussion by pullmann4rent** [supabase/supabase#28460] [Questions]
Hello,

I use supabase and have a own backend and I make my queries only at the serverside so my keys goes not in clientside and nobody knows the keys. It is still dangerous when I am not using RLS?

**Comment by GaryAustin1** [comment]
If you don't do auth, realtime or anything else ~serverside~ **browserside** and you never plan to then you are technically OK.  Unlike service_role there is no checking in Github for accidental release of anon (assuming that still works).

**Reply by pullmann4rent** [reply]
Hi thanks for your answer, I do auth but also only on serverside so nothing on client side. What could be also dangerous ?

**Reply by GaryAustin1** [reply]
That was a typo... I should have said anything else browser side...

**Comment by tweinreich** [comment]
Hi @pullmann4rent,

Generally there is nothing dangerous about it, you can see RLS as an additional insurance to protect access to resources on the DB layer.

However this leaves the task to secure accessing your DB completely up to you.

I'll make it more clear:
- Users access your app from the frontend (probably after logging in, obtaining a JWT or similar)
- Your backend fetches and stores entities from and into your DB using a user which cannot be restricted using RLS (e.g. `postgres`, which bypasses RLS per default)
- Now its your task to ensure that user A can only fetch entities he should be able to access inside your business logic, so you have to do something like this:

```java
public UserProfile getUserProfile(UUID userId) {
    Authentication currentUserAuth = SecurityContextHolder.getContext().getAuthentication();
    
    if (!userId.equals(currentUserAuth.getId()) || currentUserAuth.getRole().equals(UserRole.ADMIN)) {
        throw new UnauthorizedAccessException();
    }
    
    return userProfileRepository.findById(userId);
}
```

This is a matter of taste and it comes down to you preferring to handle this in the DB or application layer.

If you do either of these thorougly, you'll be safe 😃
